crates/ty_python_semantic/src/types/diagnostic.rs

```diff
@@ -151,7 +202,15 @@ declare_lint! {
     /// Checks for class definitions with a cyclic inheritance chain.
     ///
     /// ## Why is it bad?
-    /// TODO #14889
+    /// A cyclic inheritance chain is most likely to be an error,
+    /// as at runtime such chains must be resolvable.
+    ///
+    /// ## Examples
+    /// ```python
+    /// # foo.pyi
+    /// class A(B): ...
+    /// class B(A): ...
+    /// ```
```

Comment:
```suggestion
    /// Although forward references are natively supported in stub files,
    /// inheritance cycles are still disallowed, as it is impossible to
    /// resolve a consistent [method resolution order] for a class that
    /// inherits from itself.
    ///
    /// ## Examples
    /// ```python
    /// class A(B): ...
    /// class B(A): ...
    /// ```
    ///
    /// [method resolution order]: https://docs.python.org/3/glossary.html#term-method-resolution-order
```